ksql-execution/src/main/java/io/confluent/ksql/execution/materialization/MaterializationInfo.java

```diff
@@ -91,37 +96,43 @@ private Builder(String stateStoreName, LogicalSchema stateStoreSchema) {
     /**
      * Adds an aggregate map transform for mapping final result of complex aggregates (e.g. avg)
      *
-     * @param aggregatesInfo descriptor of aggregation functions.
+     * @param aggregator descriptor of aggregation functions.
      * @param resultSchema   schema after applying aggregate result mapping.
      * @return A builder instance with this transformation.
      */
-    public Builder mapAggregates(AggregatesInfo aggregatesInfo, LogicalSchema resultSchema) {
-      transforms.add(new AggregateMapInfo(aggregatesInfo));
+    public Builder mapAggregates(
+        final KudafAggregator aggregator,
```

Comment:
Yeah, I did think of this.   At the moment the `MaterializationInfo` tracks the `SelectValueMapper`, the `SqlPredicate` and the `KudafAggregator` for the `project`, `filter`and `mapAggregates` methods, respectively.  

However, we could easily simplify these to just a `map` and a `filter` functions that just took standard java types. Both the `project` and `mapAggregates` would then just be a `map` call and the `project` a call to `filter`.  

What do you reckon?

Reply:
I like just having map/filter. That's a more general API, and `mapAggregates` isn't that descriptive anyway.

Out of scope for this change, but maybe we can just build up a java stream and drop all this `Transform/TransformInfo/Visitor` stuff. Each transform can just be a `Supplier<Stream<GenericRow>>`, e.g.:

```
class MaterializationInfo {
    class Builder {
        Function<Stream<GenericRow>, Stream<GenericRow>> transform = s -> s;
        ...
        public Builder map(Function<GenericRow, GenericRow> f) {
            final Function<Stream<GenericRow>, Stream<GenericRow>> current = transform;
            transform = s -> current.apply(s).map(f);
        }
    }
}
```

Might not be the best for performance/debugability though.

Reply:
Cool, I'll switch.

Not sure the pure java stream/function this will work, (though I like it), as we need to also pass around the key, not just the value.